### PR TITLE
[Fix-10795][plugin] sqoop task query columns bug

### LINKED ISSUE #10795: [Bug] [plugin] sqoop task query columns config bug
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

when I submited a sqoop task which use  `All Columns` of the column type, but the sqoop task had the different columns.

Because I used to write those columns into the input box of `Some Columns`, I forgot to clear that.   

Front-end transmits data to the back-end. 

Maybe front-end also need to fix it.

image

image

### What you expected to happen

When I choose the `All Columns`, sqoop task use the all columns from the table which I choosed. 
When I choose the `Some Columns`, sqoop task use the columns from what I customized.

### How to reproduce

I will submit a pr.

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by CallMeKingsley97**
## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

fix: #10795

### CODE REVIEW

**Comment by SbloodyS** [comment]
Hi @CallMeKingsley97 , please associate the issue.

**Comment by CallMeKingsley97** [comment]
@SbloodyS  hi, there is no issue for this bug, Do I need to submit an issue?

**Comment by SbloodyS** [comment]
> @SbloodyS hi, there is no issue for this bug, Do I need to submit an issue?

Yes. To target specific issue from PR, we recommend use keyword fix: #issue_id or close: #issue_id or closes: #issue_id in your PR describe(not title, just desc). It would not only connect issue to PR but also close issue automatically when PR is be closed.

**Comment by CallMeKingsley97** [comment]
> 

hello, I associated the issue.

**Comment by SbloodyS** [comment]
> hello, I associated the issue.

I'll review this pr later. Thanks.

File: dolphinscheduler-task-plugin/dolphinscheduler-task-sqoop/src/main/java/org/apache/dolphinscheduler/plugin/task/sqoop/generator/sources/MySQLSourceGenerator.java [hunk 1]
```diff
@@ -89,9 +90,11 @@ public String generate(SqoopParameters sqoopParameters, SqoopTaskExecutionContex
                                 .append(SPACE).append(sourceMysqlParameter.getSrcTable());
                         }
 
-                        if (StringUtils.isNotEmpty(sourceMysqlParameter.getSrcColumns())) {
-                            mysqlSourceSb.append(SPACE).append(COLUMNS)
-                                .append(SPACE).append(sourceMysqlParameter.getSrcColumns());
+                        if(sourceMysqlParameter.getSrcColumnType() == SqoopColumnType.SOME_COLUMNS.getCode()){
+                            if (StringUtils.isNotEmpty(sourceMysqlParameter.getSrcColumns())) {
```
Comment (zhuxt2015) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-sqoop/src/main/java/org/apache/dolphinscheduler/plugin/task/sqoop/generator/sources/MySQLSourceGenerator.java]: Please merge two if statement

Comment (CallMeKingsley97) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-sqoop/src/main/java/org/apache/dolphinscheduler/plugin/task/sqoop/generator/sources/MySQLSourceGenerator.java]: ok, done 😀@zhuxt2015

**Comment by zhongjiajie** [comment]
@CallMeKingsley97 should we also need to make some change of our frontend code?

**Comment by CallMeKingsley97** [comment]
Yep, when user change the Column Type from `Some Columns` to `All Columns` , frontend should not transmit the data of `Some Columns` to the backend.   more infomation you can see the issue https://github.com/apache/dolphinscheduler/issues/10795
@zhongjiajie

**Comment by SbloodyS** [comment]
please check the fail CI, I found that the checkstyle fail. @CallMeKingsley97

**Comment by CallMeKingsley97** [comment]
> please check the fail CI, I found that the checkstyle fail. @CallMeKingsley97

Is that ok? 😂

**Review by SbloodyS — DISMISSED** [review]
LGTM. Thanks.

**Comment by caishunfeng** [comment]
@CallMeKingsley97 I found no modification of the front end part, so this is a full pr, can this issue be fixed?

**Comment by CallMeKingsley97** [comment]
> @CallMeKingsley97 I found no modification of the front end part, so this is a full pr, can this issue be fixed?

ok

**Comment by EricGao888** [comment]
@CallMeKingsley97 May I ask whether there are any updates for this PR?

**Comment by CallMeKingsley97** [comment]
> @CallMeKingsley97 May I ask whether there are any updates for this PR?

Completed！

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java [hunk 2]
```diff
@@ -784,6 +785,16 @@ public void checkSerialProcess(ProcessDefinition processDefinition) {
...
+        //write the parameters of the nextProcessInstance to command
+        if (StringUtils.isNotEmpty(nextProcessInstance.getCommandParam())) {
+            Map<String, String> commandStartParamsMap = JSONUtils.toMap(nextProcessInstance.getCommandParam());
+            if (MapUtils.isNotEmpty(commandStartParamsMap)) {
+                Map<String, String> paramsMap = JSONUtils.toMap(commandStartParamsMap.get(CMD_PARAM_START_PARAMS));
+                if (MapUtils.isNotEmpty(paramsMap)) {
+                    cmdParam.put(CMD_PARAM_START_PARAMS, JSONUtils.toJsonString(paramsMap));
+                }
+            }
+        }
```
Comment (ruanwenjun) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java]: Please revert this.

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java [hunk 3]
```diff
@@ -774,7 +775,8 @@ private Boolean checkCmdParam(Command command, Map<String, String> cmdParam) {
         CommandType commandTypeIfComplement = getCommandTypeIfComplement(processInstance, command);
         // reset global params while repeat running and recover tolerance fault process is needed by cmdParam
         if (commandTypeIfComplement == CommandType.REPEAT_RUNNING ||
-                commandTypeIfComplement == CommandType.RECOVER_TOLERANCE_FAULT_PROCESS) {
+                commandTypeIfComplement == CommandType.RECOVER_TOLERANCE_FAULT_PROCESS ||
+                commandTypeIfComplement == CommandType.RECOVER_SERIAL_WAIT) {
```
Comment (ruanwenjun) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java]: Please revert this.

File: dolphinscheduler-task-plugin/dolphinscheduler-task-sqoop/src/main/java/org/apache/dolphinscheduler/plugin/task/sqoop/SqoopColumnType.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dolphinscheduler.plugin.task.sqoop;
+
+public enum SqoopColumnType {
+
+    ALL_COLUMNS(0, "GET_ALL_COLUMNS"),
+    SOME_COLUMNS(1, "CUSTOMIZE_COLUMNS");
```
Comment (ruanwenjun) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-sqoop/src/main/java/org/apache/dolphinscheduler/plugin/task/sqoop/SqoopColumnType.java]:
```suggestion
    CUSTOMIZE_COLUMNS(1, "CUSTOMIZE_COLUMNS");
```

File: dolphinscheduler-task-plugin/dolphinscheduler-task-sqoop/src/main/java/org/apache/dolphinscheduler/plugin/task/sqoop/generator/sources/MySQLSourceGenerator.java [hunk 5]
```diff
@@ -89,7 +90,8 @@ public String generate(SqoopParameters sqoopParameters, SqoopTaskExecutionContex
                                     .append(SPACE).append(sourceMysqlParameter.getSrcTable());
                         }
 
-                        if (StringUtils.isNotEmpty(sourceMysqlParameter.getSrcColumns())) {
+                        if (sourceMysqlParameter.getSrcColumnType() == SqoopColumnType.SOME_COLUMNS.getCode()
```
Comment (ruanwenjun) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-sqoop/src/main/java/org/apache/dolphinscheduler/plugin/task/sqoop/generator/sources/MySQLSourceGenerator.java]:
```suggestion
                        if (sourceMysqlParameter.getSrcColumnType() == SqoopColumnType. CUSTOMIZE_COLUMNS.getCode()
```

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java (hunk 2 above)
Comment (CallMeKingsley97) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java]: done

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java (hunk 3 above)
Comment (CallMeKingsley97) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java]: done

**Review by ruanwenjun — DISMISSED** [review]
LGTM

**Comment by CallMeKingsley97** [comment]
> LGTM

Can this PR be merged now？🤭

**Review by SbloodyS — DISMISSED** [review]
+1

**Comment by SbloodyS** [comment]
Please run `mvn spotless:apply` to fix format error. @CallMeKingsley97

**Comment by CallMeKingsley97** [comment]
> Please run `mvn spotless:apply` to fix format error. @CallMeKingsley97

Done

**Review by SbloodyS — APPROVED** [review]

**Pull request merged**